subsys/net/lib/openthread/platform/radio.c

```diff
@@ -384,7 +384,9 @@ void platformRadioProcess(otInstance *aInstance)
 		while ((tx_pkt = (struct net_pkt *)k_fifo_get(&tx_pkt_fifo,
 							      K_NO_WAIT))
 		      != NULL) {
-			openthread_handle_frame_to_send(aInstance, tx_pkt);
+			if (!IS_ENABLED(CONFIG_OPENTHREAD_COPROCESSOR_RCP)) {
```

Comment:
> > > Otherwise I get a No pkt available message. Any idea, why 4 isn't enough? Is there some net packet allocated, which does not get freed?
> > 
> > 
> > I'm guessing that in RCP configuration the host can perhaps send a burst of 802.15.4 messages at once (each message needs a single net_pkt)? In NCP we still used OT stack on the NCP side, and the stack waits for the 802.15.4 packet to be transmitted before sending another one to the driver.
> > I doubt there's a memory leak - if that was the case, it would fail with 8 net_pkt's as well sooner or later.
> 
> True, but my assumption was, that we are still half-using the net stack. And this half-usage creates some packets that are no unref'ed.

On the second thought, there actually could be a leak here (in case app tried to send something over OT interface - I don't think it's actually happening, but would be good to verify). It's needed to `net_pkt_unref(tx_pkt);`  here in RCP configuration.

Reply:
Verified and when unref'ing in RCP, it works with 
```
CONFIG_NET_PKT_RX_COUNT=4
CONFIG_NET_PKT_TX_COUNT=4
```